web/src/js/ducks/modes/utils.ts

```diff
@@ -0,0 +1,94 @@
+import { getMode as getRegularModeConfig } from "./regular";
+import { getMode as getLocalModeConfig } from "./local";
+import { fetchApi } from "../../utils";
+
+export interface ModeState {
+    active: boolean;
+    name: string;
```

Comment:
I missed this in the initial review: name is not state, so it shouldn't be in state. :) Let's remove that here and make it explicit in the getMode implementation, we need to do that anyhow when we get to modes with arguments.